res/controllers/Pioneer-HID-unlock-scripts.js

```diff
@@ -0,0 +1,286 @@
+//     let losslessXor = PioneerUnlock.U32Math.losslessXor;
+
+//     const bytes = new Uint8Array(buff);
+//     return bytes.reduce((hash, val) =>
+//         (losslessMult(losslessXor(val, hash), Oxl`1000193`)),
+//         Oxl`811c9dc5`);
+// }
+
+
+PioneerUnlock.U32Math.FNVhash = function(buff) {
+    const bytes = new Uint8Array(buff);
+    return bytes.reduce((hash, val) =>
+        // appending >>> 0 to interpret result as Uint32
+        (Math.imul((val ^ hash) >>> 0, 0x1000193) & 0xFFFFFFFF) >>> 0,
+    0x811c9dc5);
+};
+
+
+// PioneerUnlock.ManufacturerUnlockSecretHiLo = {
+//     PioneerDJ: PioneerUnlock.U32Math.Oxl`680131FB`,
+//     Serato: PioneerUnlock.U32Math.Oxl`0D6F55AB`,
+//     NativeInstruments: PioneerUnlock.U32Math.Oxl`8C5B3F5D`,
+//     Atomix: PioneerUnlock.U32Math.Oxl`97779123`,
+//     Algoriddim: PioneerUnlock.U32Math.Oxl`384B522B`,
+//     Mixvibes: PioneerUnlock.U32Math.Oxl`9EA1A8B6`,
+//     Numark: PioneerUnlock.U32Math.Oxl`889127C7`,
+//     Dolby: PioneerUnlock.U32Math.Oxl`7A2B59AB`
+// };
+PioneerUnlock.ManufacturerUnlockSecretArray = {
+    PioneerDJ: [0x68, 0x01, 0x31, 0xFB],
+    Serato: [0x0D, 0x6F, 0x55, 0xAB],
+    NativeInstruments: [0x8C, 0x5B, 0x3F, 0x5D],
+    Atomix: [0x97, 0x77, 0x91, 0x23],
+    Algoriddim: [0x38, 0x4B, 0x52, 0x2B],
+    Mixvibes: [0x9E, 0xA1, 0xA8, 0xB6],
+    Numark: [0x88, 0x91, 0x27, 0xC7],
+    Dolby: [0x7A, 0x2B, 0x59, 0xAB]
+};
+
+
+
+PioneerUnlock.DeviceManufacturerDeviceID = {
+    NativeInstruments: {
+        CDJ2000NXS2: [0x02, 0xF2, 0xF4, 0x51, 0x0E, 0xD3, 0x0A, 0x2C, 0x10, 0x42],
+        DJM900NXS2: [0x43, 0xF4, 0xCD, 0xEE, 0x2C, 0x07, 0xC9, 0x59, 0x07, 0x42],
+        CDJTOUR1: [0x76, 0x8D, 0xCE, 0xDA, 0x0D, 0xD7, 0xF0, 0x10, 0x43, 0x4E],
+        DJMTOUR1: [0x50, 0x28, 0x76, 0x73, 0x8F, 0x3D, 0xC7, 0xCF, 0x1D, 0xA7],
+        XDJ1000MK2: [0xAB, 0xA7, 0x78, 0xD3, 0xB5, 0x06, 0xE2, 0x2E, 0xBB, 0xAE],
+        DJM450: [0x08, 0xEF, 0x3F, 0x2F, 0x1E, 0x7A, 0x90, 0x17, 0xF6, 0xAF],
+        DJM250MK2: [0xD2, 0xF5, 0x8B, 0x61, 0x2C, 0x62, 0x06, 0x81, 0x91, 0x39],
+        DJM750MK2: [0x86, 0x6E, 0x33, 0xBD, 0x04, 0x85, 0x2E, 0x71, 0xED, 0x21],
+        DJMV10: [0x3C, 0x83, 0xDB, 0x25, 0x9C, 0x76, 0xFE, 0x8D, 0xB2, 0xAB]
+    },
+    Serato: {
+        CDJ2000NXS2: [0xC0, 0xBD, 0x94, 0x0F, 0xB7, 0x00, 0xBB, 0x8F, 0xC7, 0x5B],
+        DJM900NXS2: [0xD5, 0x57, 0xCE, 0x8F, 0x5E, 0xF0, 0x7B, 0xC2, 0x7E, 0x98],
+        DJMV10: [0x1D, 0x3E, 0xAC, 0x55, 0xC6, 0x13, 0xD2, 0x97, 0x62, 0x3D],
+        XDJZX: [0x2D, 0x99, 0x85, 0x8D, 0x0A, 0x5A, 0x91, 0x3D, 0x4E, 0xBD],
+    },
+    PioneerDJ: {
+        CDJ2000NXS2: [0x02, 0xF2, 0xF4, 0x51, 0x0E, 0xD3, 0x0A, 0x2C, 0x10, 0x42],
+    }
+};
+
+PioneerUnlock.seedA = [0x00, 0x00, 0xFF, 0xFF, 0x00, 0x00, 0xFF, 0xFF];
+
+PioneerUnlock.keepAliveTimer = 0;
+
+PioneerUnlock.incomingData = function(data, _length) {
+
+    const manufacturer = "NativeInstruments";
+    // TODO identify correct device properly
```

Comment:
We don't have vendor ID and product ID accessible in JS domain.

Reply:
Not necessary I think. I can read that from the handshake with a little extra effort. I was just too lazy to implement it right now, I don't want the extra complexity until this has been confirmed working with a device just hardcoded.